docs/src/main/asciidoc/hibernate-search-guide.adoc

```diff
@@ -0,0 +1,627 @@
+import java.util.Objects;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+
+import io.quarkus.hibernate.orm.panache.PanacheEntity;
+
+@Entity
+public class Author extends PanacheEntity { // <1>
+
+    public String firstName;
+
+    public String lastName;
+
+    @OneToMany(mappedBy = "author", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER) // <2>
+    public List<Book> books;
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Author)) {
+            return false;
+        }
+
+        Author other = (Author) o;
+
+        return Objects.equals(id, other.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return 31;
+    }
+}
+----
+<1> We are using Hibernate ORM with Panache, it is not mandatory.
+<2> We are loading these elements eagerly so that they are present in the JSON output.
+
+[source,java]
+----
+package org.acme.hibernate.search.elasticsearch.model;
+
+import java.util.Objects;
+
+import javax.json.bind.annotation.JsonbTransient;
+import javax.persistence.Entity;
+import javax.persistence.ManyToOne;
+
+import io.quarkus.hibernate.orm.panache.PanacheEntity;
+
+@Entity
+public class Book extends PanacheEntity {
+
+    public String title;
+
+    @ManyToOne
+    @JsonbTransient // <1>
+    public Author author;
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Book)) {
+            return false;
+        }
+
+        Book other = (Book) o;
+
+        return Objects.equals(id, other.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return 31;
+    }
+}
+----
+<1> We mark this property with `@JsonbTransient` to avoid infinite loops when serializing with JSON-B.
+
+Now that we are all set with our two Hibernate ORM entities, we can start the interesting part.
+
+== Initializing the REST service
+
+While everything is not yet set up for our REST service, we can initialize it with the standard CRUD operations we will need.
+
+Just copy this content in the `LibraryResource` file created by the Maven `create-project` command:
+
+[source,java]
+----
+package org.acme.hibernate.search.elasticsearch;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.transaction.Transactional;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.acme.hibernate.search.elasticsearch.model.Author;
+import org.acme.hibernate.search.elasticsearch.model.Book;
+
+@Path("/library")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
```

Comment:
It's good practice with Quarkus (it limits the number of providers included, otherwise, you have auto-negotiation enabled and it includes all the providers in the image).